examples/todoapp/src/components/IDActionCell.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button } from "@chakra-ui/react";
+import {
+  TableCellComponentProps,
+  useFabrixClient,
+  useFabrixContext,
+} from "@fabrix-framework/fabrix";
+import { useCallback, useMemo, useState } from "react";
+
+export const IDActionCell = {
+  name: "IDActionCell",
+  type: "tableCell",
+  component: (props: TableCellComponentProps) => {
+    const client = useFabrixClient();
+    const context = useFabrixContext();
+    const [isMutating, setMutating] = useState(false);
+    const values = useMemo(() => {
+      const label = props.userProps?.["label"] ?? "Action";
+      const color = props.userProps?.["color"] ?? "blue";
+      const mutation = props.userProps?.["mutation"];
+      if (!mutation) {
+        throw new Error("Mutation is required");
+      }
+
+      return {
+        label,
+        color,
+        mutation,
+      };
+    }, [props]);
```

Comment:
These props are constants and do not change.
Shouldn't they be mapped outside the component, rather than using useMemo?

Reply:
Right. Removed `useMemo` from here.

Reply:
I assumed the following.
```tsx
type IDActionCellProps = {
  label: string;
  color: string;
  mutation: string;
};
export const IDActionCell<IDActionCellProps> = {
  name: "IDActionCell",
  type: "tableCell",
  // map TableCellComponentProps to IDActionCellProps
  props: ({ userProps }: TableCellComponentProps) => ({
    label: userProps?.label ?? "Action",
    color: props.userProps?.color ?? "blue",
    mutation: props.userProps!.mutation, // mutation is required 
  }),
  component: ({ label, color, mutation }: IDActionCellProps) => {
    const client = useFabrixClient();
    const context = useFabrixContext();
    const [isMutating, setMutating] = useState(false);
    ...
```

Because “mutation” is required, I used a non-null assertion.
I think this is equivalent to the following code, which causes a runtime error in the intermediate component.
```tsx
if (!mutation) {
throw new Error("Mutation is required");
}
```

However, your changes are fine for now.